motr/st/utils/helper.c

```diff
@@ -115,7 +116,7 @@ static int alloc_vecs(struct m0_indexvec *ext, struct m0_bufvec *data,
 		m0_indexvec_free(ext);
 		return rc;
 	}
-	rc = m0_bufvec_alloc(attr, block_count, 1);
+	rc = m0_bufvec_alloc(attr, block_count, sizeof(struct m0_md5_inc_context_pi));
```

Comment:
All attr should be allocated this way. You can create generic function to get attribute size and everywhere that function should be used, by that we can check for various different types of Data Integrity e.g. checksum, CRC etc

Reply:
done